glow/event_log.go

```diff
@@ -0,0 +1,112 @@
+package glow
+
+// This file contains a simple event logger, designed to collect a limited
+// number of timestamped logs and dump them to file or terminal.
+
+import (
+	"container/list"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+)
+
+// Event.
+type event struct {
+	ts   time.Time
+	line string
+}
+
+// Options for the event log.
+type EventLogOptions struct {
+	Expiry     time.Duration // Older events will be removed. Default is no expiry.
+	LimitBytes int           // Limit the size of events to collect. Default is not to collect events.
+}
+
+// Event log.
+type EventLog struct {
+	Options   EventLogOptions
+	Logs      *list.List // List of events
+	sizeBytes int        // Current size of events
+	mu        sync.Mutex
+}
+
+// Create a new event log.
+// The default behavior is to never expire, and keep unlimited logs.
+func NewEventLog(opts ...EventLogOptions) *EventLog {
+	var elo EventLogOptions
+	if len(opts) == 0 {
+		elo = EventLogOptions{}
+	} else {
+		elo = opts[0]
+	}
+	return &EventLog{
+		Options: elo,
+		Logs:    list.New(),
+	}
+}
+
+// Expire logs.
+func (l *EventLog) Expire(now time.Time) {
```

Comment:
A more thorough comment might be something like "Expire will discard old event log messages without logging them or  persisting them anywhere"